src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -178,12 +181,22 @@ class Container extends React.Component<Props, ComponentState> {
                     title: nextProps.intl.formatMessage({
                         id: FM.ENTITYCREATOREDITOR_TITLE_EDIT,
                         defaultMessage: 'Edit Entity'
-                    })
+                    }),
+                    enumValues: this.initEnumValues(nextProps.entity.enumValues)
                 })
             }
         }
     }
 
+    initEnumValues(enumValues: CLM.EnumValue[] | undefined): (CLM.EnumValue | null)[] {
+        if (!enumValues) {
+            return Array(CLM.MAX_ENUM_VALUES).fill(null)
+        }
+        const enumClone: CLM.EnumValue[] = JSON.parse(JSON.stringify(enumValues))
```

Comment:
Something weird is going on here. 

On the top the type is:
`enumValues: CLM.EnumValue[] | undefined`
then you clone it an caste to:
`enumClone: CLM.EnumValue[]`
but then later have to check that it's undefined
`...enumClone || []`

Seems like the middle cast should preserve what the input was `CLM.EnumValue[] | undefined` because if `enumClone` was truly an array it could never be `null` or `undefined` to trigger the `||`

Reply:
Good catch.  Left over from refactor.